doc/guides/modules.rst

```diff
@@ -10,14 +10,342 @@ system those are called *modules*. These modules must be integrated with the
 Zephyr build system, as described in more detail in other sections on
 this page.
 
-Zephyr depends on several categories of modules, including:
+To be classified as a candidate for being included in the default list of
+modules, an external project is required to have its own life-cycle outside
+the Zephyr Project, that is, reside in its own repository, and have its own
+contribution and maintainance workfow and release process. Zephyr modules
+should not contain code that is written exclusively for Zephyr.
+
+  .. note::
+
+     Code which is written for Zephyr and works exclusively with Zephyr
+     should not be submitted as a Zephyr module; instead, it should be
+     part of the main Zephyr tree.
+
+Modules to be included in the default manifest of the Zephyr project need to
+provide functionality or features endorsed and approved by the project technical
+steering committee and should comply with the
+:ref:`module licensing requirements<modules_licensing>` and
+:ref:`contribute guidelines<modules_contributing>`. They should also have a
+Zephyr developer that is commited to maintain the module code base.
+
+Zephyr depends on several categories of modules, including but not limited to:
 
 - Debugger integration
 - Silicon vendor Hardware Abstraction Layers (HALs)
 - Cryptography libraries
 - File Systems
 - Inter-Process Communication (IPC) libraries
 
+This page summarizes a list of policies and best practices which aim at
+better organizing the workflow in Zephyr modules.
+
+Module Repositories
+*******************
+
+* All modules included in the default manifest shall be hosted in repositories
+  under the Zephyr Project Github organization.
+
+* Module repository names should follow the convention of using lowercase
+  letters and dashes instead of underscores. This rule will apply to all
+  new module repositories.
+
+  .. note::
+
+     It is not required to modify the names of existing module repositories
+     to make them comply with the above naming convention.
+
+* If the module has an external (upstream) project repository, the module
+  repository should preserve the upstream repository folder structure.
+
+* Modules should use "zephyr" as the default name for the repository main
```

Comment:
If you are going to fork modules as proposed here (which I would certainly try to avoid), a requirement should be to not  fork branches present in the upstream repository to avoid confusion.
(I do not see any problem with having replicas of upstream branches, but I would not keep a "master" in a module which has either diverged or is fully decoupled from the upstream "master")

Reply:
I would say that some module might be pure mirror of its upstream - why to enforce zephyr branch then?

Reply:
@nvlsianpu @aescolar this was also discussed online. We favor having a common branch name for the default branch in module repositories. The enforced uniformity makes it easy for developers to browse the code in zephyr modules, and for maintainers and release engineers to perform updates in module repositories.

It is uncommon for modules to be pure mirrors of upstream, but even if they are, there is no particular reason why the main branch should be other than "zephyr".